cytodynamics-nucleus/src/main/java/com/linkedin/cytodynamics/nucleus/ParentRelationship.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright 2018-2019 LinkedIn Corporation
+ * All Rights Reserved.
+ *
+ * Licensed under the BSD 2-Clause License (the "License").
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.cytodynamics.nucleus;
+
+import java.util.Set;
+
+
+/**
+ * Contains objects and configuration for an individual parent classloading step for {@link IsolatingClassLoader}.
+ */
+public class ParentRelationship {
+  private final ClassLoader parentClassLoader;
+  private final IsolationLevel isolationLevel;
+  private final Set<GlobMatcher> parentPreferredClassPatterns;
+  private final Set<GlobMatcher> blacklistedClassPatterns;
+  private final Set<GlobMatcher> whitelistedClassPatterns;
+
+  public ParentRelationship(ClassLoader parent,
```

Comment:
Do we want this to be constructable directly? If not, we could make it package private, to encourage the use of the builder.